src/trossen_rslidar_node.cpp

```diff
@@ -0,0 +1,249 @@
+
+  // input related
+  get_parameter<uint16_t>("msop_port", driver_parameters_.input_param.msop_port);
+  get_parameter<uint16_t>("difop_port", driver_parameters_.input_param.difop_port);
+  get_parameter<std::string>("host_address", driver_parameters_.input_param.host_address);
+  get_parameter<std::string>("group_address", driver_parameters_.input_param.group_address);
+  get_parameter<bool>("use_vlan", driver_parameters_.input_param.use_vlan);
+  get_parameter<std::string>("pcap_path", driver_parameters_.input_param.pcap_path);
+  get_parameter<float>("pcap_rate", driver_parameters_.input_param.pcap_rate);
+  get_parameter<bool>("pcap_repeat", driver_parameters_.input_param.pcap_repeat);
+  get_parameter<uint16_t>("user_layer_bytes", driver_parameters_.input_param.user_layer_bytes);
+  get_parameter<uint16_t>("tail_layer_bytes", driver_parameters_.input_param.tail_layer_bytes);
+
+  std::string lidar_type;
+  get_parameter<std::string>("lidar_type", lidar_type);
+  driver_parameters_.lidar_type = strToLidarType(lidar_type);
+
+  // decoder
+  get_parameter<bool>("wait_for_difop", driver_parameters_.decoder_param.wait_for_difop);
+  get_parameter<bool>("use_lidar_clock", driver_parameters_.decoder_param.use_lidar_clock);
+  get_parameter<float>("min_distance", driver_parameters_.decoder_param.min_distance);
+  get_parameter<float>("max_distance", driver_parameters_.decoder_param.max_distance);
+  get_parameter<float>("start_angle", driver_parameters_.decoder_param.start_angle);
+  get_parameter<float>("end_angle", driver_parameters_.decoder_param.end_angle);
+  get_parameter<bool>("dense_points", driver_parameters_.decoder_param.dense_points);
+  get_parameter<bool>("ts_first_point", driver_parameters_.decoder_param.ts_first_point);
+
+  // mechanical decoder
+  get_parameter<bool>("config_from_file", driver_parameters_.decoder_param.config_from_file);
+  get_parameter<std::string>("angle_path", driver_parameters_.decoder_param.angle_path);
+
+  uint16_t split_frame_mode;
+  get_parameter<uint16_t>("split_frame_mode", split_frame_mode);
+  driver_parameters_.decoder_param.split_frame_mode = SplitFrameMode(split_frame_mode);
+
+  get_parameter<float>("split_angle", driver_parameters_.decoder_param.split_angle);
+  get_parameter<uint16_t>("num_blks_split", driver_parameters_.decoder_param.num_blks_split);
+
+  // transform
+  get_parameter<float>("x", driver_parameters_.decoder_param.transform_param.x);
+  get_parameter<float>("y", driver_parameters_.decoder_param.transform_param.y);
+  get_parameter<float>("z", driver_parameters_.decoder_param.transform_param.z);
+  get_parameter<float>("roll", driver_parameters_.decoder_param.transform_param.roll);
+  get_parameter<float>("pitch", driver_parameters_.decoder_param.transform_param.pitch);
+  get_parameter<float>("yaw", driver_parameters_.decoder_param.transform_param.yaw);
+
+  driver_parameters_.input_type = InputType::ONLINE_LIDAR;
+
+//   driver_parameters_.print();
+
```

Comment:
```suggestion
```

Reply:
@lukeschmitt-tr refer comment in trossen_lidar_main about print commands.

Reply:
Let's keep this one, to print if a parameter print_config is true (from config file). Just to leverage a function they have already created inside the driver.

Reply:
Works for me

Reply:
Are we going to uncomment this line then?